wg-platform/src/std_lib/udp_reader.rs

```diff
@@ -0,0 +1,36 @@
+use std::io;
+use std::net::UdpSocket;
+use std::sync::Weak;
+
+use wg_traits::Endpoint;
+use wg_traits::udp::Reader;
+
+use super::{get_connection_aborted_err, udp_endpoint::StdUdpEndpoint};
+
+pub struct StdUdpReader {
+    socket: Weak<UdpSocket>,
+}
```

Comment:
I wonder if we even need the Weak/Arc:
If we did not close the socket on drop, but did it explicitly, then when we write to/read from the socket, the OS just returns an error to us and we can simply pass it on.